promql/promqltest/testdata/histograms.test

```diff
@@ -625,9 +690,15 @@ load_with_nhcb 5m
 eval instant at 50m histogram_quantile(0.99, {__name__=~"request_duration_seconds\\d*_bucket"})
   expect fail
 
+eval instant at 50m histogram_quantiles({__name__=~"request_duration_seconds\\d*_bucket"}, "q", 0.99)
+  expect fail
+
 eval instant at 50m histogram_quantile(0.99, {__name__=~"request_duration_seconds\\d*"})
     expect fail
 
+eval instant at 50m histogram_quantiles({__name__=~"request_duration_seconds\\d*"}, "q", 0.99)
+    expect fail
```

Comment:
There is also another case where `histogram_quantiles()` might break timeseries uniqueness – if the quantile label overwrites some existing label. E.g.:

```
some_histogram{q="a"} ...
some_histogram{q="b"} ...

histogram_quantiles(some_histogram, "q", 0)
```

Looking at the code, I believe this should be caught here:

https://github.com/prometheus/prometheus/blob/1ce84d8e2f576474cf9cd956ebb1d50f2a333977/promql/engine.go#L1963-L1965

Though maybe it would be worth adding a corresponding testcase, just to be sure?